pkg/scheduler/worker/workerpool_test.go

```diff
@@ -56,19 +68,19 @@ func TestWorkerPool(t *testing.T) {
 		require.NoError(t, inventory.Delete(clusterState.Cluster.RuntimeID))
 	}()
 
+	//create test invoker to be able to verify invoker calls
+	testInvoker := &testInvoker{}
+
 	//create reconciliation for cluster
-	reconRepo := reconciliation.NewInMemoryReconciliationRepository()
-	reconEntity, err := reconRepo.CreateReconciliation(clusterState, nil)
+	testInvoker.reconRepo = reconciliation.NewInMemoryReconciliationRepository()
```

Comment:
If you managed to remove reconRepo out of testInvoker, we don't have to change this test completely. Normally for the existing test, we should be really careful when changing it. If the related logic is not changed, the test should be prevented to change.

Unit tests is our first line of defense to save us to sleep well, keep it stable:)

Reply:
Unfortunately, we need to reconRepo in the Invoker ti update the Status of the operations. The old Unit test did not cover such scenario and thus needed to be adapted